libs/ast-references/src/types/contract_reference.rs

```diff
@@ -0,0 +1,147 @@
+/**
+ * ContractReference.rs
+ * Definition of ContractReference struct
+ * author: 0xMemoryGrinder
+*/
+
+use std::fmt;
+use std::hash::{Hash, Hasher};
+use std::rc::Rc;
+use std::cell::RefCell;
+use crate::types::location::Location;
+use crate::types::file_reference::FileReference;
+use crate::types::struct_reference::StructReference;
+use crate::types::function_reference::FunctionReference;
+use crate::types::property_reference::PropertyReference;
+
+/******************************************************************************
+ *                                  Types                                     *
+ *****************************************************************************/
+
+pub struct ContractReference {
+    pub name: String,
+    pub location: Location,
+    pub file: Rc<RefCell<FileReference>>,
+    pub structs: Vec<Rc<RefCell<StructReference>>>,
+    pub functions: Vec<Rc<RefCell<FunctionReference>>>,
+    pub properties: Vec<Rc<RefCell<PropertyReference>>>,
+}
```

Comment:
Add events, errors, modifiers, version.
Need to handle constants in properties

Reply:
The version does not need to be handled by this crate. Its only motive is to simplify the references traversal for the LSP extension.